extensions/amp-story/0.1/page-scaling.js

```diff
@@ -0,0 +1,253 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {Services} from '../../../src/services';
+import {dev, user} from '../../../src/log';
+import {dict} from '../../../src/utils/object';
+import {isExperimentOn} from '../../../src/experiments';
+import {childElementsByTag, matches} from '../../../src/dom';
+import {px, setImportantStyles} from '../../../src/style';
+import {renderAsElement} from './simple-template';
+import {throttle} from '../../../src/utils/rate-limit';
+import {toArray} from '../../../src/types';
+
+
+/** @private @const {number} */
+const MIN_LAYER_WIDTH_PX = 380;
```

Comment:
How did we pick these min/max values?  Is there a way to keep these in sync with the actual min/max widths that we allow in our UIs?  e.g. I *think* the mobile UI can go larger than 520 before it hits the desktop breakpoint, and I think it can go smaller than 380 as well.

Reply:
Ignore this; I got it backwards.  I thought we were scaling within the range min...max, but we're scaling *outside* the range min...max.